tests/hil/tests/settings/test.c

```diff
@@ -85,10 +98,49 @@ void hil_test_entry(const struct golioth_client_config *config)
 
     status = golioth_settings_register_int(settings, "TEST_WRONG_TYPE", on_test_int, NULL);
 
+    status = golioth_settings_register_bool(settings, "TEST_CANCEL", on_test_cancel, NULL);
+
     (void) status;
+}
+
+void hil_test_entry(const struct golioth_client_config *config)
+{
+    connected_sem = golioth_sys_sem_create(1, 0);
+    cancel_all_sem = golioth_sys_sem_create(1, 0);
+
+    golioth_debug_set_cloud_log_enabled(false);
+
+    struct golioth_client *client = golioth_client_create(config);
+    golioth_client_register_event_callback(client, on_client_event, NULL);
+
+    golioth_sys_sem_take(connected_sem, GOLIOTH_SYS_WAIT_FOREVER);
+
+    perform_settings_registration(client);
```

Comment:
If we change this function to return `struct golioth_settings *` then we can keep `settings` as a local stack variable instead of moving it to a global.